src/topology/continuous_function/units.lean

```diff
@@ -0,0 +1,97 @@
+/-
+Copyright (c) 2022 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import topology.continuous_function.compact
+import analysis.normed_space.units
+import algebra.algebra.spectrum
+
+/-!
+# Units of continuous functions
+
+This file concerns itself with `C(α, β)ˣ` and `C(α, βˣ)` when `α` is a topological space
+and `β` has an appropriate algebraic and topological structure.
```

Comment:
I would personally prefer to have slightly more descriptive variable names, that emphasize that `α` is a random space (e.g. `X`?) whereas `β` has a multiplicative structure (e.g. `M` or `R`?).

Reply:
Note that the type class assumption on `α`, `β` are very general, only that both are topological spaces and `has_zero β`. In the algebraic structure section, the type class assumptions are continually varying, sometimes additive and sometimes multiplicative (in general, I have to do them separately and can't use `to_additive` because all the multiplicative structure has `with_zero` or some such). Also, this matches the `topology/continuous_function/bounded` file. Given the above, if you really want I can change it (but please provide a bit more specifics about exactly what you would like if so), but I think it makes more sense this way.

Reply:
How about `X` instead of `α`, and then use `M` for monoids and `R` for rings?